medallion/views/objects.py

```diff
@@ -50,6 +49,102 @@ def get_custom_headers(api_root, id_):
     return headers
 
 
+def get_and_enforce_limit(api_root, id_, objects):
+    """
+    Defines TAXII API - Pagination:
+    Pagination section (3.5 <link here>`__), Get Object Manifests section (5.3 <link here>`__), and Get Objects section (5.4 <link here>`__)
+
+    Args:
+        api_root (str): the base URL of the API Root
+        id_ (str): the `identifier` of the Collection being requested
+        objects (dict): objects being returned for request (may have been filtered by other query parameters)
+
+    Returns:
+        headers:
+            header values matching objects being sent in server response
+        objects:
+            Values matching server request, sorted by ascending date_added
+
+    """
+    headers = {}
+    if request.args.get('limit'):
+        limit = int(request.args['limit'])
+    else:
+        limit = current_app.taxii_config["max_page_size"]
+    try:
+        manifest = current_app.medallion_backend.get_object_manifest(
+            api_root, id_, None, ("id",),
+        )
+        if manifest:
```

Comment:
This block is not meant to be executed if `limit` is not provided in the request and instead can use the short version present in get_custom_headers()

Reply:
Are you talking about the whole function? The values still need to get sorted by date_added, which happens here.

Reply:
You could modify this function, but currently that means that objects["more"] = True would be a false statement if I made a regular request (without `limit`)

Reply:
After the most recent push, objects["more"] seems to work as intended, only being set as true if not all of the objects were returned. objects["more"] is returned as false when there is no limit parameter or when the limit value does not affect the total number of objects returned.

Reply:
Yep, confirmed its working for non-paginated requests and when limit == the result set.